doc/whats_new/v1.2.rst

```diff
@@ -250,10 +250,15 @@ Changelog
   matrices in a variety of estimators and avoid an `EfficiencyWarning`.
   :pr:`23139` by `Tom Dupre la Tour`_.
 
-- |Fix| :class:`NearestCentroid` now raises an informative error message at fit-time
-  instead of failing with a low-level error message at predict-time.
+- |Fix| :class:`neighbors.NearestCentroid` now raises an informative error message
+  at fit-time instead of failing with a low-level error message at predict-time.
```

Comment:
I see why this should be fixed, but can you revert this? We do a sweep of the whats new before release to fix these types of issues.

This way, the PR is focused on addressing the issue in the title. When we merge the title of the PR becomes the commit message. By keeping the PR related to the title, each commit is more self contained. Also, less changes usually means it is easier to review, which means the PR will get merge faster.

Reply:
Woops, yes, I was actually hesitating to modify it for the very reasons you mentioned 😅